### DISCUSSION TITLE
Versions

### DISCUSSION THREAD

**Discussion by FrankC01** [ChainMovers/suibase#8] [General]
Currently, the `workdirs`, have specific "branch" sources The reality is that you (user) may want/need the current definition PLUS a `sui genesis` from the `main` branch of the repo (which is by definition a bit more unstable)

The user may also want various `...net` versions, e.g.:
- devnet-lastest
- devnet-0.27.0
- devnet-0.26.0
- etc.

**Comment by mario4tier** [comment]
About various version (one at the time)
---
(using "localnet" because the only one implemented for now)

The ~/sui-base/workdirs/localnet/sui-repo is a "welcome open" place where user can git control the version.

They can even work on modif there.

"$ localnet update"... is problematic though. Proposed fix:

- "$ localnet update" should fail if the sui-repo is not on the expected "devnet" branch OR have modifications. Inform the user that it must first switch to the expected branch, stash the changes etc...

About various version (multiple co-existing)
---
Is there a use case for multiple localnet workdir at the same time? 

I will twist that question from a different angle:  is there a use case for selecting a "localnet2" instance for the SDKs and other tools? Probably no. 

Yet, some "super user" might need to have multiple repo co-located... and they can.  But only one "~/sui-base/workdirs/localnet/sui-repo" really matter... after all... only one localnet can run at the time.

This is where I was thinking the "super user" may decide to "override" a sui-repo location with a "sui-base.yaml".

Essentially, it makes "~/sui-base/workdirs/localnet/sui-repo" to symlink to any external repo.

**Comment by FrankC01** [comment]
Yes, right now I have 0.27.0 localnet in `sui-base` but am using a VM to setup an 0.28.0 localnet to develop to the changes coming.

**Reply by mario4tier** [reply]
Seems we are on the same page (although I see now that my question might have cause a miscommunication).

The options I am considering are:

(1) the user override the "sui-repo" under a single localnet workdir
    VERSUS 
(2) Creating multiple localnet workdir

... and I am leaning toward (1) because then the "localnet" script is still working...

Will try your use case today (simulate override with main branch) and see how it goes.

In the end, multiple repo can co-exist for localnet, but only one has to be involve when "$ localnet start" is done.

**Reply by FrankC01** [reply]
Just to reemphasize `localnet start 0.27.1` and `localnet start 0.28.0` was the consideration.
In the `localnet` script, search for the version tag before downloading (of course)
